DB/src/main/java/io/deephaven/db/v2/locations/local/ReadOnlyParquetTableLocation.java

```diff
@@ -181,15 +197,22 @@ private GroupingFile getGroupingFile(String name) {
                             break;
                         case BINARY:
                         case FIXED_LEN_BYTE_ARRAY:
+                            //noinspection rawtypes
+                            final ObjectCodec codec;
                             if (isCodec) {
                                 final String codecParams = keyValueMetaData.get(ParquetTableWriter.CODEC_ARGS_PREFIX + name);
-                                //noinspection rawtypes
-                                final ObjectCodec codec = CodecCache.DEFAULT.getCodec(codecName, codecParams);
-                                //noinspection unchecked
-                                toPage = ToObjectPage.create(dataType, codec, columnChunkReader.getDictionary());
+                                codec = CodecCache.DEFAULT.getCodec(codecName, codecParams);
                             } else {
-                                throw new TableDataException("No codec in parquet file for binary blob.");
+                                final String codecParams;
+                                if (typeName == PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY) {
+                                    codecParams = Integer.toString(type.getTypeLength());
+                                } else {
+                                    codecParams = null;
+                                }
+                                codec = CodecCache.DEFAULT.getCodec(io.deephaven.util.codec.ByteArrayCodec.class.getName(), codecParams);
```

Comment:
`io.deephaven.util.codec.ByteArrayCodec#ByteArrayCodec` actually seems to kind of suck for reading data we didn't write. It has some highly-questionable null handling, etc.
I think you should write a trivial new `PassThroughByteArrayCodec` (or similar name) that does a no-op decode. It might still be worthwhile to take an expected width argument, and maybe even enforce it, but that's as far as I would go towards allowing arguments.
I'll note that we always use `Binary.getBytes()` in the reading code (`io.deephaven.db.v2.locations.parquet.topage.ToObjectPage`), and that this always returns a fresh copy of the byte array that thus belongs to the caller. The [JavaDoc](https://www.javadoc.io/doc/org.apache.parquet/parquet-column/1.8.1/org/apache/parquet/io/api/Binary.html#getBytes()) doesn't say so, but there's a clear contrast with [getBytesUnsafe](https://www.javadoc.io/doc/org.apache.parquet/parquet-column/1.8.1/org/apache/parquet/io/api/Binary.html#getBytesUnsafe()) and I've read all the impls in our library.

Reply:
Take a peek at SimpleByteArrayCodec.